tracer.go

```diff
@@ -25,6 +25,10 @@ type Options struct {
 	TrimUnsampledSpans bool
 	// Recorder receives Spans which have been finished.
 	Recorder SpanRecorder
+	// An EventManager can be used to enhance the tracer by effectively
+	// attaching external code to trace events. See NetTraceIntegrator
+	// for a practical example.
+	EventManager func() func(Event)
```

Comment:
agree that `NewSpanEventListener` is a mouthful... the thing I wanted to get across is that – though the `Options` struct is per-`Tracer`, the listener is scoped to the lifetime of a `Span`. That's pretty essential to how it works in practice and is not clear at the moment (IMO).